### PR TITLE
Report diagnostic for extra tokens following an expression

### LINKED ISSUE #126: slide "1 2" -/> no panic
Expected `slide "1 2"` to output

```
no panic
```

Actual output:

```
thread 'main' panicked at 'assertion failed: self.done()', libslide/src/parser/expression_parser.rs:85:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This issue was created for a failed [slide command expectation][1].

[Command run output][2]

[1]: https://github.com/ayazhafiz/slide/issues/56#issuecomment-646374361
[2]: https://github.com/ayazhafiz/slide-workflows/actions/runs/140310575

### PR DESCRIPTION
**Pull request by ayazhafiz**
Currently we panic, but slide should just report an error instead.

Closes #126

### CODE REVIEW

File: slide/src/test/ui/extra_items_expr_pat.slide [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+~~~stderr
+error: Unexpected extra tokens
+  |
+1 |   _a + $b -#c + 1 2
+  |  _________________^
+2 | | 3 4 5 6 / 7 ^ 8
+3 | |   9 
+  | |____^ not connected to a primary expression
+  |
```
Comment (luke-bhan) [inline slide/src/test/ui/extra_items_expr_pat.slide:22]: this error msg is slighly confusing because it connects the two lines.

Comment (ayazhafiz) [reply slide/src/test/ui/extra_items_expr_pat.slide:22]: i agree, it might be nicer if we underline the whole thing. I'll look into it..

**Comment by ayazhafiz** [comment]
LGTM

**Comment by ayazhafiz** [comment]
lgtm

**Comment by ayazhafiz** [comment]
lgtm

**Pull request merged**